src/java/org/apache/cassandra/schema/Schema.java

```diff
@@ -81,6 +115,37 @@ private Schema()
         }
     }
 
+    /**
+     * Add entries to system_schema.* for the hardcoded system keyspaces
+     * 
+     * See CASSANDRA-16856/16996. Make sure schema pulls are synchronized to prevent concurrent schema pull/writes
+     */
+    public static synchronized void saveSystemKeyspacesSchema()
+    {
+        SchemaKeyspace.saveSystemKeyspacesSchema();
+    }
+
+    /**
+     * See CASSANDRA-16856/16996. Make sure schema pulls are synchronized to prevent concurrent schema pull/writes
+     */
+    public static synchronized void truncateSystemKeyspacesSchema()
+    {
+        SchemaKeyspace.truncate();
+    }
+
+    /**
+     * See CASSANDRA-16856/16996. Make sure schema pulls are synchronized to prevent concurrent schema pull/writes
+     */
+    public static synchronized void convertSystemKeyspacesSchemaToMutations()
```

Comment:
nit: `convertSystemKeyspacesSchemaToMutations()` -> `systemKeyspaceAsMutations()`

Reply:
`SchemaPullVerbHandler` will call this version, right? (...and then we can remove `synchronized` from `SchemaKeyspace.convertSchemaToMutations()`?)

Reply:
Apologies that sync is an overlooked left-over.